### PR TITLE
Improve `pydantic/_internal/_fields.py` docstring

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: pydantic/_internal/_fields.py [hunk 1]
```diff
@@ -22,6 +22,19 @@ def get_type_hints_infer_globalns(
     localns: dict[str, Any] | None = None,
     include_extras: bool = False,
 ) -> dict[str, Any]:
+    """Gets type hints for an object by inferring the global namespace.
+
+    It uses the `typing.get_type_hints`, The only thing that we do here is fetching
+    global namespac from `obj.__module__` if it is not `None`.
```
Comment (Viicos) [inline pydantic/_internal/_fields.py:28]: @hramezani A bit late: `namespace`

Comment (hramezani) [reply pydantic/_internal/_fields.py:28]: Thanks
PR is welcome if you would like to fix it. Otherwise If will create a PR

Comment (Viicos) [reply pydantic/_internal/_fields.py:28]: Will do

**Pull request merged**